### ISSUE TITLE
DataLossError and agent.restore_model

### ISSUE THREAD

**Issue by ycps** [reinforceio/tensorforce#229]
## Environment
python3.5 (python3.5 -m venv)
tensorflow 1.4.0 - [wheel](https://github.com/mind/wheels/releases/download/tf1.4-cpu/tensorflow-1.4.0-cp35-cp35m-linux_x86_64.whl)

## Minimal working example
To easily reproduce the issue, edit [tensorforce/examples/quickstart.py](https://github.com/reinforceio/tensorforce/blob/master/examples/quickstart.py).
After the agent definition, add following lines:
```python
agent.save_model(directory = 'parameters/prefix', append_timestep = False)
agent.restore_model(file = 'parameters/checkpoint')
```

## Output log
The following output is trimmed in `[…]`:
```
[…]
2017-11-13 14:45:20.359162: W tensorflow/core/util/tensor_slice_reader.cc:95] Could not open /home/data/parameters/checkpoint: Data loss: not an sstable (bad magic number): perhaps your file is in a different file format and you need to use a different restore operator?
2017-11-13 14:45:20.359706: W tensorflow/core/util/tensor_slice_reader.cc:95] Could not open /home/data/parameters/checkpoint: Data loss: not an sstable (bad magic number): perhaps your file is in a different file format and you need to use a different restore operator?
2017-11-13 14:45:20.359721: W tensorflow/core/framework/op_kernel.cc:1192] Data loss: Unable to open table file /home/data/parameters/checkpoint: Data loss: not an sstable (bad magic number): perhaps your file is in a different file format and you need to use a different restore operator?
Traceback (most recent call last):
  File "/home/env/lib64/python3.5/site-packages/tensorflow/python/client/session.py", line 1323, in _do_call
    return fn(*args)
  File "/home/env/lib64/python3.5/site-packages/tensorflow/python/client/session.py", line 1302, in _run_fn
    status, run_metadata)
  File "/home/env/lib64/python3.5/site-packages/tensorflow/python/framework/errors_impl.py", line 473, in __exit__
    c_api.TF_GetCode(self.status.status))
tensorflow.python.framework.errors_impl.DataLossError: Unable to open table file /home/data/parameters/checkpoint: Data loss: not an sstable (bad magic number): perhaps your file is in a different file format and you need to use a different restore operator?
	 [[Node: save/RestoreV2_1 = RestoreV2[dtypes=[DT_FLOAT], _device="/job:localhost/replica:0/task:0/device:CPU:0"](_arg_save/Const_0_0, save/RestoreV2_1/tensor_names, save/RestoreV2_1/shape_and_slices)]]
[…]
  File "/home/env/src/tensorforce/tensorforce/models/model.py", line 309, in setup
    save_relative_paths=True
  File "/home/env/lib64/python3.5/site-packages/tensorflow/python/training/saver.py", line 1218, in __init__
    self.build()
  File "/home/env/lib64/python3.5/site-packages/tensorflow/python/training/saver.py", line 1227, in build
    self._build(self._filename, build_save=True, build_restore=True)
  File "/home/env/lib64/python3.5/site-packages/tensorflow/python/training/saver.py", line 1263, in _build
    build_save=build_save, build_restore=build_restore)
  File "/home/env/lib64/python3.5/site-packages/tensorflow/python/training/saver.py", line 751, in _build_internal
    restore_sequentially, reshape)
  File "/home/env/lib64/python3.5/site-packages/tensorflow/python/training/saver.py", line 427, in _AddRestoreOps
    tensors = self.restore_op(filename_tensor, saveable, preferred_shard)
  File "/home/env/lib64/python3.5/site-packages/tensorflow/python/training/saver.py", line 267, in restore_op
    [spec.tensor.dtype])[0])
  File "/home/env/lib64/python3.5/site-packages/tensorflow/python/ops/gen_io_ops.py", line 1021, in restore_v2
    shape_and_slices=shape_and_slices, dtypes=dtypes, name=name)
  File "/home/env/lib64/python3.5/site-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "/home/env/lib64/python3.5/site-packages/tensorflow/python/framework/ops.py", line 2956, in create_op
    op_def=op_def)
  File "/home/env/lib64/python3.5/site-packages/tensorflow/python/framework/ops.py", line 1470, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

DataLossError (see above for traceback): Unable to open table file /home/data/parameters/checkpoint: Data loss: not an sstable (bad magic number): perhaps your file is in a different file format and you need to use a different restore operator?
	 [[Node: save/RestoreV2_1 = RestoreV2[dtypes=[DT_FLOAT], _device="/job:localhost/replica:0/task:0/device:CPU:0"](_arg_save/Const_0_0, save/RestoreV2_1/tensor_names, save/RestoreV2_1/shape_and_slices)]]
```

**Comment by michaelschaarschmidt** [comment]
Thanks, we will look into it soon

**Comment by slundell** [comment]
This is to do with the logic in agent.restore_model wich is a bit convoluted to me. Restoring the model with a full path seems to work just fine:
`agent.save_model(directory='parameters/prefix', append_timestep = False)
agent.model.restore(file='parameters/prefix')`

**Comment by ycps** [comment]
Fix suggested in #232 solved the issue for me.

**Comment by michaelschaarschmidt** [comment]
#232 fix in master now, I will close this for now but re-open it if you run into more issues
